documentation/dataplane/dash-routing-actions.md

```diff
@@ -0,0 +1,197 @@
+
+1. [Overview](#overview)
+2. [Stage transition actions](#stage-transition-actions)
+   1. [Basic stage transition actions](#basic-stage-transition-actions)
+      1. [`drop` action](#drop-action)
+      2. [`trap` action](#trap-action)
+   2. [Matching stage transition actions](#matching-stage-transition-actions)
+      1. [IP-based stage transition actions](#ip-based-stage-transition-actions)
+         1. [`lpmrouting` action](#lpmrouting-action)
+         2. [`maprouting` action](#maprouting-action)
+      2. [Port-based stage transition actions](#port-based-stage-transition-actions)
+         1. [`portmaprouting` action](#portmaprouting-action)
+3. [Packet transformation actions](#packet-transformation-actions)
+   1. [`staticencap` action](#staticencap-action)
+   2. [`tunnel` action](#tunnel-action)
+   3. [`4to6` action](#4to6-action)
+   4. [`6to4` action](#6to4-action)
+   5. [`nat` action](#nat-action)
+
+## Overview
+
+On high level, there are 2 types of routing actions:
+
+- Packet transformation actions:
+  - These actions are used to transform the packet header fields, such as modify the packet fields, adding encaps, etc.
+  - These actions can only be used in the routing types that specified in the `routing_type` field in each match stage entry.
+- Stage transition actions:
+  - These actions are used to describe the stage transitions in the DASH pipeline, such as transit to maprouting stage.
+  - This actions can only be used in the routing types that specified in the `transition` field in each match stage entry.
+
+For how routing action works, please refer to the ["Routing action and routing types" section in DASH packet flow HLD](../general/dash-sai-pipeline-packet-flow.md#57-routing-actions-and-routing-types).
+
+## Stage transition actions
+
+### Basic stage transition actions
+
+#### `drop` action
+
+- Actions:
+  - Drop the packet after the entry is matched.
+
+#### `trap` action
+
+- Actions:
+  - Trap the packet to CPU after the entry is matched.
+
+### Matching stage transition actions
+
+All matching stage transition actions will support the following parameters:
+
+- `default_routing_type`: If no entry is found, use this routing type to route the packet. If `default_routing_type` is not set, the packet will be dropped by default.
+- `stage_index`: If there are multiple stage of the same stage type, use this field to specify which stage to transit to. The index is starting from 0 and by default 0.
+
+#### IP-based stage transition actions
+
+All IP-based stage transition actions will support the following parameters:
+
+- `use_src_ip`: Use source IP in routing and mapping stages.
+- `ip_mask`: IP mask to apply before matching the entries.
+
+##### `lpmrouting` action
+
+- Actions:
+  - The packet will be routed to the next stage based on the LPM routing table.
+
+The port mapping entries can be described as below:
```

Comment:
Typo. You probably want to mention lpmrouting entries instead of "port mapping" entries.

Reply:
Right, it is a typo. I will rephase this sentence and update again.

Reply:
fixed.